capstone/src/main/webapp/script.js

```diff
@@ -110,25 +122,207 @@ async function getAnalysis(text, lang, requestedAttributes, tokenizer) {
+  const words = getSubstrings(substring, TokenizerEnum['WORD']);
+  for (let i = 0; i < words.length; i++) {
+    const wordEl = createAnyElement('span', words[i]);
+    const bareWord = words[i].replace(/[\s.,\/#!$%\^&\*;:{}=\-_`~()]/g, ''); // Remove any punctuation & whitespace
+    wordEl.onclick = function() { setUpReplacements(substring, lang, bareWord); }
+    wordEl.classList.add('output-word'); 
+    substringEl.appendChild(wordEl);
+  }
   colorSubstring(substringEl, toxicityScore);
      
   // Attach a tooltip (info-box for the substring)
   const tooltipEl = createTooltip(attributes);
   substringEl.appendChild(tooltipEl);
   result.appendChild(substringEl);
 }
+ 
+/** Sets up the replacements section */
+function setUpReplacements(text, lang, substringForReplacements) {
+  // Clear previous results
+  document.getElementById('perspective-datamuse-chart').innerHTML = '';
+  document.getElementById('perspective-datamuse-extras').innerHTML = '';
+
+  document.getElementById('about-container').style.display = 'block';
+
+  getReplacements(text, lang, substringForReplacements);
+  showDataMuseWordReplacementOptions(text, lang, substringForReplacements);
+}
+
+/** Gets the replacements & their score changes for a subtring based on the Datamuse API */
+async function getReplacements(text, lang, substringForReplacements) {
+  // Clear previous results
+  const analysisContainer = document.getElementById('perspective-datamuse-analysis');
+  analysisContainer.innerHTML = '';
+  document.getElementById('perspective-datamuse-chart').innerHTML = '';
+
+  // Set up loading spinner
+  const loadingEl = document.createElement('p');
+  loadingEl.className = 'spinner-border';
+  analysisContainer.appendChild(loadingEl);
+
+  // Get the word type
+  let wordType = getReplacementsWordType();
+  if (!wordType) {
+    wordType = DATAMUSE_ATTRIBUTES['Means like'];
+  }
+
+  // Get the toxicity score of original string 
+  const toxicityOfOriginal = await getOriginalToxicity(text, lang, analysisContainer, loadingEl);
+  if (!toxicityOfOriginal) {
+    analysisContainer.removeChild(loadingEl)
+    analysisContainer.appendChild(createAnyElement('b', 'Perspective API was not able to score the original sentence'));
+    return;
+  }
+  
+  // Get Datamuse API replacements of substring
+  const replacements = await callDatamuse(substringForReplacements, wordType);
+  if (replacements.length === 0) {
+    analysisContainer.removeChild(loadingEl)
+    analysisContainer.appendChild(createAnyElement('b', 'Datamuse API did not find any words or phrases matching your query'));
+    return;
+  }
+ 
+  printReplacements(text, substringForReplacements, replacements, toxicityOfOriginal, lang, analysisContainer, loadingEl);
+}
+
+/** Prints the modified strings to the page with their Perspective TOXICITY scorings */
+async function printReplacements(text, substringForReplacements, replacements, toxicityOfOriginal, lang, container, loadingEl) {
```

Comment:
nit: I'd suggest something like `showReplacements` or `renderReplacements`. `print` makes me think of printing to the console or actually printing.

Reply:
I chose `renderReplacements`